src/unix.rs

```diff
@@ -17,17 +17,19 @@ use std::thread::{self, Builder, JoinHandle};
 use std::time::Duration;
 
 #[derive(Debug)]
-pub enum Client {
-    /// `--jobserver-auth=R,W`
-    Pipe { read: File, write: File },
-    /// `--jobserver-auth=fifo:PATH`
-    Fifo {
-        file: File,
-        path: PathBuf,
-        /// it can only go from false -> true but not the other way around, since that
-        ///  could cause a race condition.
-        is_non_blocking: AtomicBool,
-    },
+enum ClientCreationArg {
+    Fds { read: c_int, write: c_int },
```

Comment:
Was thinking about `BorrowedFd<'static>` here to introduce niche for enum optimization:

```suggestion
    Fds { read: BorrowedFd<'static>, write: BorrowedFd<'static> },
```

But I was afraid that it would introduce more problem, as now fd unsoundness becomes a potentially lifetime unsoundness issue, so I didn't do it.

Reply:
~~Or we could simply store a `Box<str>` here instead, since it is only used in `string_arg()` and returns a `&str` anyway.~~

Not possible as we actually care which variant is used here for `configure`.